hotspot/src/share/vm/gc_implementation/g1/dirtyCardQueue.cpp

```diff
@@ -291,10 +313,11 @@ void DirtyCardQueueSet::concatenate_logs() {
   for (JavaThread* t = Threads::first(); t; t = t->next()) {
     DirtyCardQueue& dcq = t->dirty_card_queue();
     if (dcq.size() != 0) {
-      void **buf = t->dirty_card_queue().get_buf();
+      void **buf = dcq.get_buf();
```

Comment:
The 11u version changes this to `void** buf`. Was there a reason for the difference here?

Reply:
I think the return type of get_buf which  is defined for the first time is inappropriate, see [here](https://github.com/openjdk/jdk8u-dev/blob/f4260a4deed5210fc56f08774e8ca50cf6f7cc8e/hotspot/src/share/vm/gc_implementation/g1/dirtyCardQueue.hpp#L74), first commited in [here](https://github.com/openjdk/jdk8u-dev/commit/18f3386a98e67f1c5b75c4cc47a0ebb898c59b9b). 
The 11u version changes the buf allocation by  `BufferNode::make_buffer_from_node`(see [here](https://github.com/openjdk/jdk11u/blob/master/src/hotspot/share/gc/g1/dirtyCardQueue.cpp#L178)), however the jdk8 does not do that.
I can modify it to `void** buf`, should  i do that?